src/rrules.jl

```diff
@@ -0,0 +1,37 @@
+############################ AD rules ############################################
+function rrule(F::judiPropagator{T, O}, m::AbstractArray{T}, q::AbstractArray{T}) where {T, O}
+    y = F(m, q)
+    dims = length(q)
+    function pullback(Δy)
+        dm = @thunk(∇m(F, m, q, Δy))
+        dq = @thunk(∇source(F, m, Δy, dims))
+        return (NoTangent(), dm, dq)
```

Comment:
I personally would move line 6 and 7 into 8

Reply:
Mmuch more readable like that no